src/configLoader.js

```diff
@@ -1,7 +1,10 @@
 import fs from 'fs';
-import mapDeep from 'deepdash/mapDeep';
 import path from 'path';
 import yaml from 'js-yaml';
+import { getEnvironmentVarPointers, lookUpEnvironmentVar } from './environment';
+import jsonPointer from 'json-ptr';
+
+const GLOBAL_ENV_VAR_REGEX = /\$\{(.+?)\}/g;
```

Comment:
This could be tied to the other regex by making it `new RegExp(ENV_VAR_REGEX.source, 'g')`

Reply:
Although now that I say that, I'm not completely following why we want the global flag in one case, and not in the other. Could you explain for me?

Reply:
yes, and that's because it's not a good reason 😄 javascript global regexes have an extremely inconvenient feature that effectively couples them to specific strings: https://stackoverflow.com/questions/1520800/why-does-a-regexp-with-global-flag-give-wrong-results. they store the state of the last match's index and try to match the rest of the string (including for a new string!!). so, when looping through the JSON pointers looking for environment variables, a global regex will fail non-deterministically, unless we clone it each iteration.

a brief example:

<img width="339" alt="KongfigEnvSubstitution" src="https://user-images.githubusercontent.com/3241493/68308899-9a6c7d00-007b-11ea-9780-7d279f5a1246.png">

so yes, i'll make that change

Reply:
thanks, i hate it!!

Reply:
I take it `replace` doesn't use `lastIndex`? otherwise, why isn't this a problem for the actual substitution function?

Reply:
`replace` does do the global replace. i think using the definition in `environment` and transforming it into a global regex here, as you originally suggested, will clear that up

Reply:
![Screen Shot 2019-11-06 at 9 33 33 AM](https://user-images.githubusercontent.com/473542/68312362-8f641d80-0078-11ea-9c44-20496c209131.png)
this is what I meant. I thought maybe `replace` would set `lastIndex` the way test does. seems like it doesn't, which is why the global flag doesn't cause issues in this context.

Reply:
oh yeah, makes sense. i hadn't checked that explicitly but kind of assumed it was true based on the `replace` docs and behavior. it doesn't make any sense to me when global regexes behave differently from regular ones in javascript... similar thing with `match` and `matchAll` (which isn't really compatible with... anything yet)